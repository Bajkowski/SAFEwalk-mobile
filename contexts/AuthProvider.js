import React, { createContext, useReducer, useMemo } from "react";
import { AsyncStorage } from "react-native";

export const AuthContext = createContext();

export function AuthProvider({ children }) {
  const [state, dispatch] = useReducer(authReducer, {
    // initial states
    isLoading: true,
    isSignout: false,
    userToken: null,
    userType: "user",
    userEmail: null
  });

  // Handle Login
  async function login(userType, { email, password }) {
    try {
      // Put code below for this: make login GET request to server

      // For now, while server is not set, use email as value for user token
      await AsyncStorage.setItem("userToken", JSON.stringify(email));
      // Store the user type i.e. which type of user is logged in, user or SAFEwalker.
      await AsyncStorage.setItem("userType", userType);
      // Store the user email
      await AsyncStorage.setItem("userEmail", JSON.stringify(email));

      dispatch({
        type: "LOG_IN",
        token: JSON.stringify(email),
        userType: userType,
        userEmail: email
      });
    } catch (error) {
      throw new Error("Error in login(): " + error);
    }
  }

  // Handle signout
  async function signout() {
    try {
      await AsyncStorage.removeItem("userToken");
      await AsyncStorage.removeItem("userType");
      await AsyncStorage.removeItem("userEmail");

      dispatch({ type: "SIGN_OUT" });
    } catch (error) {
      throw new Error("Error in signout(): " + error);
    }
  }

  // Handle register
  async function register({
    email,
    password,
    firstName,
    lastName,
    phoneNumber
  }) {
    try {
      // Put code below for this: make register POST request to server

      // For now, while server is not set, use email as value for user token
      await AsyncStorage.setItem("userToken", JSON.stringify(email));
      // Store the user type i.e. which type of user is logged in, user or SAFEwalker.
      await AsyncStorage.setItem("userType", userType);
      // Store the user email
      await AsyncStorage.setItem("userEmail", JSON.stringify(email));

      dispatch({
        type: "LOG_IN",
        token: JSON.stringify(email),
        userType: "user",
        userEmail: email
      });
    } catch (error) {
      throw new Error("Error in register(): " + error);
    }
  }

  const authContextValue = useMemo(() => {
    return {
      isLoading: state.isLoading,
      isSignout: state.isSignout,
      userToken: state.userToken,
      userType: state.userType,
      userEmail: state.userEmail,
      dispatch,
      login,
      signout,
      register
    };
  }, [state]);

  return (
    <AuthContext.Provider value={authContextValue}>
      {children}
    </AuthContext.Provider>
  );
}

function authReducer(prevState, action) {
  switch (action.type) {
    case "RESTORE_TOKEN":
      return {
        ...prevState,
        isLoading: false,
        userToken: action.token,
        userType: action.userType,
        userEmail: action.userEmail
      };
    case "LOG_IN":
      return {
        ...prevState,
        isSignout: false,
        userToken: action.token,
        userType: action.userType,
        userEmail: action.userEmail
      };
    case "SIGN_OUT":
      return {
        ...prevState,
        isSignout: true,
        userToken: null,
        userType: null,
        userEmail: null
      };
    default: {
      throw new Error(`Unhandled auth action type: ${action.type}`);
    }
  }
}
