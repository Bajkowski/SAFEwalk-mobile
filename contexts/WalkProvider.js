import React, { createContext, useReducer, useMemo } from "react";
import { AsyncStorage } from "react-native";

export const WalkContext = createContext();

export function WalkProvider({ children }) {
  const [state, dispatch] = useReducer(walkReducer, {
    // initial states
    isWalkLoading: true,
    isWalkActive: "false", // has to be a string due to AsyncStorage constraints (cannot store boolean)

    // initial walk states
    walkId: null,
    startLat: null,
    startLng: null,
    destLat: null,
    destLng: null,

    // initial user state
    userEmail: null,
    walkerEmail: null,

    // initial SAFEwalker state
    userSocketId: null,
    walkerSocketId: null,
  });

  // Set up walk ID
  async function setWalkAsActive() {
    try {
      // Store walk ID
      await AsyncStorage.setItem("isWalkActive", "true");

      dispatch({
        type: "SET_WALK_ACTIVE",
        isWalkActive: "true",
      });
    } catch (error) {
      console.error("Error in setIsWalkActive(): " + error);
    }
  }

  // Set up walk ID
  async function setWalkId(walkId) {
    try {
      // Store walk ID
      await AsyncStorage.setItem("walkId", walkId);

      dispatch({
        type: "SET_WALK_ID",
        walkId: walkId,
      });
    } catch (error) {
      console.error("Error in setWalkId(): " + error);
    }
  }

  // Remove walk ID
  async function removeWalkId() {
    try {
      // remove walk ID
      await AsyncStorage.removeItem("WalkId");

      dispatch({ type: "REMOVE_WALK_ID" });
    } catch (error) {
      console.error("Error in removeWalkId(): " + error);
    }
  }

  // Store Coordinates (startLat, startLng, destLat, destLng)
  async function setCoordinates(startLat, startLng, destLat, destLng) {
    try {
      await AsyncStorage.setItem("startLat", startLat);
      await AsyncStorage.setItem("startLng", startLng);
      await AsyncStorage.setItem("destLat", destLat);
      await AsyncStorage.setItem("destLng", destLng);

      dispatch({
        type: "SET_COORDINATES",
        startLat: startLat,
        startLng: startLng,
        destLat: destLat,
        destLng: destLng,
      });
    } catch (error) {
      console.error("Error in setCoordinates(): " + error);
    }
  }

  // Store User information (walkId, email and socket ID)
  async function setUserInfo(walkId, userEmail, userSocketId) {
    try {
      await AsyncStorage.setItem("walkId", walkId);
      await AsyncStorage.setItem("userEmail", userEmail);
      await AsyncStorage.setItem("userSocketId", userSocketId);

      dispatch({
        type: "SET_USER_INFO",
        walkId: walkId,
        userEmail: userEmail,
        userSocketId: userSocketId,
      });
    } catch (error) {
      console.error("Error in seUserInfo(): " + error);
    }
  }

  // Store Safewalker information (email and socket ID)
  async function setWalkerInfo(walkerEmail, walkerSocketId) {
    try {
      await AsyncStorage.setItem("walkerEmail", walkerEmail);
      await AsyncStorage.setItem("walkerSocketId", walkerSocketId);

      dispatch({
        type: "SET_WALKER_INFO",
        walkerEmail: walkerEmail,
        walkerSocketId: walkerSocketId,
      });
    } catch (error) {
      console.error("Error in setWalkerInfo(): " + error);
    }
  }

  // remove all current walk-related information
  async function resetWalkContextState() {
    try {
      await AsyncStorage.removeItem("userEmail");
      await AsyncStorage.removeItem("userSocketId");
      await AsyncStorage.removeItem("walkerEmail");
      await AsyncStorage.removeItem("walkerSocketId");

      await AsyncStorage.removeItem("walkId");
      await AsyncStorage.removeItem("startLat");
      await AsyncStorage.removeItem("startLng");
      await AsyncStorage.removeItem("destLat");
      await AsyncStorage.removeItem("destLng");

      dispatch({ type: "RESET" });
    } catch (error) {
      console.error("Error in resetContextState(): " + error);
    }
  }

  const walkContextValue = useMemo(() => {
    return {
      // states
      isWalkLoading: state.isWalkLoading,
      isWalkActive: state.isWalkActive,
      userEmail: state.userEmail,
      walkerEmail: state.walkerEmail,
      userSocketId: state.userSocketId,
      walkerSocketId: state.walkerSocketId,
      walkId: state.walkId,
      startLat: state.startLat,
      startLng: state.startLng,
      destLat: state.destLat,
      destLng: state.destLng,

      // functions
      setWalkAsActive,
      setWalkId,
      removeWalkId,
      setCoordinates,
      setUserInfo,
      setWalkerInfo,
      resetWalkContextState,
      dispatch,
    };
  }, [state]);

  return (
    <WalkContext.Provider value={walkContextValue}>
      {children}
    </WalkContext.Provider>
  );
}

function walkReducer(prevState, action) {
  switch (action.type) {
    case "RESTORE_STATE":
      return {
        ...prevState,
        isWalkLoading: false,
        isWalkActive: action.isWalkActive,
        userEmail: action.userEmail,
        walkerEmail: action.walkerEmail,
        userSocketId: action.userSocketId,
        walkerSocketId: action.walkerSocketId,
        walkId: action.walkId,
        startLat: action.startLat,
        startLng: action.startLng,
        destLat: action.destLat,
        destLng: action.destLng,
      };
    case "SET_WALK_ACTIVE":
      return {
        ...prevState,
        isWalkActive: "true",
      };
    case "SET_WALK_ID":
      return {
        ...prevState,
        walkId: action.walkId,
      };
    case "REMOVE_WALK_ID":
      return {
        ...prevState,
        walkId: null,
      };
    case "SET_COORDINATES":
      return {
        ...prevState,
        startLat: action.startLat,
        startLng: action.startLng,
        destLat: action.destLat,
        destLng: action.destLng,
      };
    case "SET_USER_INFO":
      return {
        ...prevState,
        userEmail: action.userEmail,
        userSocketId: action.userSocketId,
        walkId: action.walkId,
      };
    case "SET_WALKER_INFO":
      return {
        ...prevState,
        walkerEmail: action.walkerEmail,
        walkerSocketId: action.walkerSocketId,
      };
    case "RESET":
      return {
        ...prevState,
        isWalkLoading: false,
        isWalkActive: "false",
        userEmail: null,
        userSocketId: null,
        walkerEmail: null,
        walkerSocketId: null,
        walkId: null,
        startLat: null,
        startLng: null,
        destLat: null,
        destLng: null,
      };

    default: {
      console.error(`Unhandled walk action type: ${action.type}`);
    }
  }
}
