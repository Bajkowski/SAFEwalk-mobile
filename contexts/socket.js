const io = require("socket.io-client");
module.exports = io("http://127.0.0.1:3000");
// module.exports = io("http://365b17d8.ngrok.io");
