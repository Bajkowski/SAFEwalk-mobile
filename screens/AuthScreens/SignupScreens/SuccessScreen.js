import React, { useState, useEffect, useContext } from "react";
import { StyleSheet, Text, View, ActivityIndicator } from "react-native";
import { Image } from "react-native-elements";
import Button from "./../../../components/Button";
import ErrorText from "./../../../components/ErrorText";
import Spacer from "../../../components/Spacer";

// Libraries
import { SafeAreaView } from "react-native-safe-area-context";
import { useForm } from "react-hook-form";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

// Constants
import colors from "./../../../constants/colors";
import url from "./../../../constants/api";
import style from "./../../../constants/style";

import { AuthContext } from "./../../../contexts/AuthProvider";

export default function SuccessSignupScreen({ route }) {
  const [isLoginError, setIsLoginError] = useState(false);
  const [isUserNotAvailable, setIsUserNotAvailable] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const { login, register } = useContext(AuthContext);

  // hide error after 5 seconds
  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsLoginError(false);
      setIsUserNotAvailable(false);
    }, 5000);

    return () => clearTimeout(timeout);
  }, [isLoginError, isUserNotAvailable]);

  // upon pressing the submit button
  function handleLogin() {
    setIsLoading(true);
    console.log("email: " + route.params.email);
    console.log("password: " + route.params.password);
    fetch(url + "/api/Login", {
      method: "GET",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json;charset=UTF-8",
        email: route.params.email,
        password: route.params.password,
        isUser: true, // since this is user login, then isUser has to be true
      },
    })
      .then((response) => response.json())
      .then((data) => {
        setIsLoading(false);
        console.log("data: " + data);
        // The endpoint only returns a string upon success
        // and a full body response if there's an error.
        // Therefore, if data.status exists, then this means it's an error.
        if (data.status) {
          console.log("data in if: " + JSON.stringify(data));
          if (data.status === 404) {
            console.log("captured 404! User not available.");
            setIsUserNotAvailable(true);
          } else {
            console.log("Unknown error " + data.status + ". Try again");
            setIsLoginError(true);
          }
        }
        // The endpoint only returns a string upon success,
        // so because of that, if it's a success, data.status would be null.
        else {
          console.log("data in else: " + data);
          console.log("email: " + route.params.email);
          login("user", data, route.params.email);
        }
      })
      .catch((error) => {
        console.log("Error in login(): " + error);
        setIsLoginError(true);
        setIsLoading(false);
      });
  }

  return (
    <SafeAreaView style={styles.container}>
      {/* Progress animation */}
      <View style={styles.progressContainer}>
        <View style={styles.progressCircleContainer}>
          <View style={styles.progressCircle}>
            <Text style={styles.progressCircleText}>✓</Text>
          </View>
          <Text style={styles.progressDescription}>Credentials</Text>
        </View>

        <View style={[styles.progressLine, { marginRight: 3 }]} />

        <View style={styles.progressCircleContainer}>
          <View style={styles.progressCircle}>
            <Text style={styles.progressCircleText}>✓</Text>
          </View>
          <Text style={styles.progressDescription}>Basic Info</Text>
        </View>

        <View
          style={[styles.progressLine, { marginLeft: 5, marginRight: 10 }]}
        />

        <View style={styles.progressCircleContainer}>
          <View style={styles.progressCircle}>
            <Text style={styles.progressCircleText}>✓</Text>
          </View>
          <Text style={styles.progressDescription}>Finish</Text>
        </View>
      </View>

      {/* Success image*/}
      <View>
        <Text style={styles.textSuccess}>Success!</Text>
        <Spacer />
        <Image
          source={require("./../../../assets/signup-success.jpg")}
          style={styles.image}
          PlaceholderContent={<ActivityIndicator />}
        />
      </View>

      {/* Button go to main screen */}
      <View style={styles.buttonContainer}>
        {isLoginError && (
          <ErrorText>
            There was an error. Please try again. (isLoginError)
          </ErrorText>
        )}
        {isUserNotAvailable && (
          <ErrorText>
            There was an error. Please try again. (isUserNotAvailable)
          </ErrorText>
        )}
        <Button
          title="Go to main screen"
          loading={isLoading}
          disabled={isLoading}
          onPress={() => handleLogin()}
        />
      </View>
      <Spacer />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    justifyContent: "space-between",
  },
  progressContainer: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  progressCircleContainer: {
    alignItems: "center",
  },
  progressCircle: {
    alignItems: "center",
    justifyContent: "center",
    borderColor: colors.darkorange,
    width: 50,
    height: 50,
    borderRadius: 50,
    borderWidth: 3,
  },
  progressCurrentCircle: {
    borderColor: colors.darkorange,
  },
  progressCircleText: {
    color: colors.darkorange,
  },
  progressDescription: {
    marginTop: 10,
  },
  progressLine: {
    borderWidth: 1,
    borderColor: colors.darkorange,
    width: 40,
    marginBottom: 20,
  },
  textSuccess: {
    alignSelf: "center",
    fontSize: hp("5%"),
  },
  image: {
    height: hp("33%"),
  },
  buttonContainer: {
    marginHorizontal: style.marginContainerHorizontal,
  },
});
