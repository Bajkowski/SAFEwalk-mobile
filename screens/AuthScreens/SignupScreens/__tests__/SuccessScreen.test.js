import React from "react";
import { create, shallow } from "react-test-renderer";
import renderer from "react-test-renderer";
import {
  fireEvent,
  NativeTestEvent,
  act,
  wait,
  waitForElement,
  waitForElementToBeRemoved,
  render,
} from "@testing-library/react-native";

import SuccessScreen from "../SuccessScreen";
import { AuthProvider } from "./../../../../contexts/AuthProvider";

jest.mock("react-native-safe-area-context", () => ({
  SafeAreaView: ({ children }) => <>{children}</>,
}));

jest.mock("@react-navigation/stack", () => {
  return { SignupStack: () => jest.fn() };
});

global.fetch = jest.fn(() => Promise.resolve());

export const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    replace: jest.fn(),
  },
  ...props,
});

let props;
let component;
let screen;

describe("SuccessScreen ", () => {
  beforeEach(() => {
    props = createTestProps();

    let route = {
      params: {
        email: "a@wisc.edu",
        password: "asdfg",
      },
    };

    component = (
      <AuthProvider>
        <SuccessScreen route={route} />
      </AuthProvider>
    );

    screen = render(component);
  });

  it("renders correctly", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays the correct loading component", () => {
    // expect(screen.getByLabelText("loadingComponent")).toBeTruthy();
    expect(screen.getAllByText("âœ“")).toBeTruthy();
    expect(screen.getByText("Credentials")).toBeTruthy();
    expect(screen.getByText("Basic Info")).toBeTruthy();
    expect(screen.getByText("Finish")).toBeTruthy();
  });

  it("has the correct title", () => {
    expect(screen.getByText("Success!")).toBeTruthy();
  });

  it("displays the SAFEwalker image", () => {
    expect(screen.getByLabelText("successImage")).toBeTruthy();
  });

  it("has the Finish button", () => {
    expect(screen.getByLabelText("Go to main screen")).toBeTruthy();
  });

  it("succeeds if email and password is valid", () => {
    let finishButton = screen.getByLabelText("Go to main screen");
    act(() => {
      fireEvent.press(finishButton);
    });
  });

  it("fails if user is not available in server (404)", async () => {
    let data = {
      status: 404,
    };

    global.fetch = jest.fn(() => Promise.resolve(data));

    let finishButton = screen.getByLabelText("Go to main screen");
    act(() => {
      fireEvent.press(finishButton);
    });

    // wait for Error appearance
    // await wait(() => {
    //   expect(screen.queryByLabelText("isUserNotAvailable")).toBeNull();
    // });
  });

  it("shows error if input is empty", async () => {
    global.fetch = jest.fn(() => Promise.reject());

    let route = {
      params: {
        email: "",
        password: "",
      },
    };

    component = (
      <AuthProvider>
        <SuccessScreen route={route} />
      </AuthProvider>
    );
    screen = render(component);

    let finishButton = screen.getByLabelText("Go to main screen");
    act(() => {
      fireEvent.press(finishButton);
    });

    // wait for Error appearance
    await wait(() => {
      expect(screen.getByLabelText("isLoginError")).toBeTruthy();
    });
  });
});
