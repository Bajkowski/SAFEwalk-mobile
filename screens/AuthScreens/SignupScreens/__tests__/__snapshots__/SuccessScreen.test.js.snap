// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessScreen  renders correctly 1`] = `
Array [
  <View
    accessibilityLabel="loadingComponent"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#C45312",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#C45312",
            }
          }
        >
          ✓
        </Text>
      </View>
      <Text
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        Credentials
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#C45312",
            "borderWidth": 1,
            "marginBottom": 20,
            "width": 40,
          },
          Object {
            "marginRight": 3,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#C45312",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#C45312",
            }
          }
        >
          ✓
        </Text>
      </View>
      <Text
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        Basic Info
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderColor": "#C45312",
            "borderWidth": 1,
            "marginBottom": 20,
            "width": 40,
          },
          Object {
            "marginLeft": 5,
            "marginRight": 10,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#C45312",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#C45312",
            }
          }
        >
          ✓
        </Text>
      </View>
      <Text
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        Finish
      </Text>
    </View>
  </View>,
  <View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontSize": 66.5,
        }
      }
    >
      Success!
    </Text>
    <View
      style={
        Array [
          Object {
            "paddingVertical": 22.5,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "position": "relative",
        }
      }
    >
      <Image
        accessibilityLabel="successImage"
        onLoad={[Function]}
        source={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 440,
              "width": undefined,
            },
          ]
        }
        testID="RNE__Image"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      />
      <View
        accessibilityElementsHidden={true}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#bdbdbd",
              "height": 440,
              "justifyContent": "center",
            }
          }
          testID="RNE__Image__placeholder"
        >
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": 440,
          }
        }
      />
    </View>
  </View>,
  <View
    style={
      Object {
        "marginHorizontal": 75,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <TouchableOpacity
        accessibilityLabel="Go to main screen"
        activeOpacity={0.3}
        disabled={false}
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fd6e11",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 80,
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 26.5,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Go to main screen
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "paddingVertical": 22.5,
        },
        undefined,
      ]
    }
  />,
]
`;
